### PR TITLE
Update document with debugging on Windows information

### PR DESCRIPTION
**Pull request by ghost**
### What does this PR do?

Update document to add Windows debugging info.

### Additional Notes

n/a

### Describe your test plan

Read the document. Not a code change.

### CODE REVIEW

**Review by olivielpeau — APPROVED** [review]
Thanks for writing this up, LGTM 👍

File: docs/dev/tools.md [hunk 1]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+## Debugging on Windows
+
+Prior to 7.23, agent binaries (Datadog Agent, Process Agent, Trace Agent, etc.) on Windows contain symbol information.
+
+Starting from 7.23, agent binaries on Windows have debugging information stripped. The original files are packed in a
+file called debug package.
+
+### Prerequisite
+
+To debug agent process, Golang Runtime, Git and Golang Delve must be installed.
```
Comment (sarina-dd) [inline docs/dev/tools.md:91]:
```suggestion
To debug Agent process, Golang Runtime, Git and Golang Delve must be installed.
```

File: docs/dev/tools.md [hunk 2]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+
+Download the matching debug package. If the MSI file is datadog-agent-7.23.0-x86_64.msi, the debug package should be
```
Comment (sarina-dd) [inline docs/dev/tools.md:93]:
```suggestion
Download the matching debug package. If the MSI file is `datadog-agent-7.23.0-x86_64.msi`, the debug package should be
```

File: docs/dev/tools.md [hunk 3]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+datadog-agent-7.23.0-x86_64.debug.zip.
```
Comment (sarina-dd) [inline docs/dev/tools.md:94]:
```suggestion
`datadog-agent-7.23.0-x86_64.debug.zip`.
```

File: docs/dev/tools.md [hunk 4]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+
+### Live Debugging
+
+Delve debugger on Windows cannot attach to the service process. The corresponding Windows service must be stopped and
+disabled.
+
+For pre 7.23, start the agent executable in the interactive session.
```
Comment (sarina-dd) [inline docs/dev/tools.md:101]:
```suggestion
For pre 7.23, start the Agent executable in the interactive session.
```

File: docs/dev/tools.md [hunk 5]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
 ```
 and running the agent in the foreground.
 
+## Debugging on Windows
+
+Prior to 7.23, agent binaries (Datadog Agent, Process Agent, Trace Agent, etc.) on Windows contain symbol information.
```
Comment (sarina-dd) [inline docs/dev/tools.md:84]:
```suggestion
Prior to 7.23, Agent binaries (Datadog Agent, Process Agent, Trace Agent, etc.) on Windows contain symbol information.
```

File: docs/dev/tools.md [hunk 6]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+
+Starting from 7.23, agent binaries on Windows have debugging information stripped. The original files are packed in a
```
Comment (sarina-dd) [inline docs/dev/tools.md:86]:
```suggestion
Starting from 7.23, Agent binaries on Windows have debugging information stripped. The original files are packed in a
```

File: docs/dev/tools.md [hunk 7]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+datadog-agent-7.23.0-x86_64.debug.zip.
+
+### Live Debugging
+
+Delve debugger on Windows cannot attach to the service process. The corresponding Windows service must be stopped and
+disabled.
+
+For pre 7.23, start the agent executable in the interactive session.
+
+For 7.23 or later version, find the file in the debug package. For agent.exe, the file in debug package is under
```
Comment (sarina-dd) [inline docs/dev/tools.md:103]:
```suggestion
For 7.23 or later version, find the file in the debug package. For `agent.exe`, the file in debug package is under
```

File: docs/dev/tools.md [hunk 8]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+\src\datadog-agent\src\github.com\DataDog\datadog-agent\bin\agent\agent.exe.debug. You might find the same file under
```
Comment (sarina-dd) [inline docs/dev/tools.md:104]:
```suggestion
`\src\datadog-agent\src\github.com\DataDog\datadog-agent\bin\agent\agent.exe.debug`. You might find the same file under
```

File: docs/dev/tools.md [hunk 9]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+\omnibus-ruby\src\cf-root\bin. Use either one is fine. Copy the file to replace the executable file you want to debug,
```
Comment (sarina-dd) [inline docs/dev/tools.md:105]:
```suggestion
`\omnibus-ruby\src\cf-root\bin`. Use either one is fine. Copy the file to replace the executable file you want to debug,
```

File: docs/dev/tools.md [hunk 10]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+start the agent executable in the interactive session.
+
+Use "dlv attach PID" to attach to the running process and start debugging.
```
Comment (sarina-dd) [inline docs/dev/tools.md:108]:
```suggestion
Use `dlv attach PID` to attach to the running process and start debugging.
```

File: docs/dev/tools.md [hunk 11]
```diff
@@ -79,6 +79,40 @@ pdb.set_trace()
...
+
+### Non-live Debugging
+
+Use "dlv core DUMPFILE EXEFILE" to debug against a dump file.
```
Comment (sarina-dd) [inline docs/dev/tools.md:112]:
```suggestion
Use `dlv core DUMPFILE EXEFILE` to debug against a dump file.
```

**Review by sarina-dd — COMMENTED** [review]
Docs editorial review completed

File: docs/dev/tools.md (hunk 11 above)
Comment (mikezhu-dd) [reply docs/dev/tools.md]: Thanks so much for the review. I've committed all changes you proposed. Thanks.

**Review by derekwbrown — APPROVED** [review]

**Comment by mikezhu-dd** [comment]
Some checks failed, but they are not related to this change (this change is doc change only). Will merge now.

**Pull request merged**